### PR TITLE
Bridges API

### PR DESCRIPTION
**Pull request by avbel**

### CODE REVIEW

File: lib/bridge.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+			path   : "bridges",
+			method : "GET",
+			qs     : params
+		})
+		.then(function (response) {
+			return response.body;
+		})
+		.asCallback(callback);
+	};
+
```
Comment (dtolb) [inline lib/bridge.js:85]: Same as domains and endpoints for pagination. Originally the page & size weren't documented. I added the parameters to the docs

File: lib/bridge.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+	/**
+	 * Gets a list of bridges.
+	 * @param {Object} params Query parameters for listing bridges
+	 * @param {Number} [params.page=0] Used for pagination to indicate the page requested for querying
```
Comment (dtolb) [inline lib/bridge.js]: We don't want to document the page parameter as we're encouraging people to use the links in the returned header. Eventually we'll move from page & size to a sort key. This way we don't have to update the SDK

File: lib/bridge.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+	/**
+	 * Speak sentence to the bridge
+	 * @param {String} bridgeId The ID of the bridge
+	 * @param {String} sentence A sentence to speak to the bridge.
+	 * @param {Object} params Optional parameters for the operation.
+	 * @param {Function} [callback] Callback for the operation
+	 * @return {BridgeResponse} A promise for the operation
+	 */
+	this.speakSentence = function (bridgeId, sentence, params, callback) {
```
Comment (dtolb) [inline lib/bridge.js]: need to handle the case where no parameters are used and a callback is passed.

``` node
bw.Bridge.speakSentence('123', 'hi how are you', function(err, res) {});
```

will throw error:

```
Unhandled rejection 
    at handleResponse (/Users/dtolbert/code/node-bandwidth/lib/client.js:21:10)
    at tryCatcher (/Users/dtolbert/code/node-bandwidth/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/Users/dtolbert/code/node-bandwidth/node_modules/bluebird/js/release/promise.js:503:31)
    at Promise._settlePromise (/Users/dtolbert/code/node-bandwidth/node_modules/bluebird/js/release/promise.js:560:18)
    at Promise._settlePromise0 (/Users/dtolbert/code/node-bandwidth/node_modules/bluebird/js/release/promise.js:605:10)
    at Promise._settlePromises (/Users/dtolbert/code/node-bandwidth/node_modules/bluebird/js/release/promise.js:684:18)
    at Promise._fulfill (/Users/dtolbert/code/node-bandwidth/node_modules/bluebird/js/release/promise.js:629:18)
    at Request._callback (/Users/dtolbert/code/node-bandwidth/node_modules/bluebird/js/release/nodeback.js:42:21)
    at Request.self.callback (/Users/dtolbert/code/node-bandwidth/node_modules/request/request.js:199:22)
    at emitTwo (events.js:106:13)
    at Request.emit (events.js:191:7)
    at Request.<anonymous> (/Users/dtolbert/code/node-bandwidth/node_modules/request/request.js:1036:10)
    at emitOne (events.js:101:20)
    at Request.emit (events.js:188:7)
    at IncomingMessage.<anonymous> (/Users/dtolbert/code/node-bandwidth/node_modules/request/request.js:963:12)
    at emitNone (events.js:91:20)
```

This is how I handled that scenario for [endpoints](https://github.com/bandwidthcom/node-bandwidth/blob/endPoints/lib/endpoint.js#L63) but if you know a better way I'm open to suggestions

Be sure to update tests as well.

File: lib/bridge.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+	 * @return {BridgeResponse} A promise for the operation
+	 */
+	this.playAudio = function (bridgeId, fileUrl, params, callback) {
+		if (!params) {
+			params = {};
+		}
+		params.fileUrl = fileUrl;
+		return audioApi(bridgeId, params).asCallback(callback);
+	};
+
```
Comment (dtolb) [inline lib/bridge.js:206]: Same as speaksentence... need to handle situation where callback is used with no parameters

**Comment by dtolb** [comment]
After talking with @nguyer we'd rather just have `playAudio` and `speakSentence` take the bridgeID and an object WITH ALL the parameters:

``` node
Bridge.playAudio('123', {fileUrl: 'fileUrl', tag: 'tag', etc: 'etc...'}, function(err,res){});
```

### OR

If you want to have a simple `playAudio` and `speakSentence` along with an advanced option. we're not opposed to:

``` node
Bridge.playAudio('123', 'fileurl', function(err,res){});

//AND
Bridge.playAudioAdvanced('123', 'fileUrl', {tag: 'tag'}, function(err,res){});

//OR
Bridge.playAudioAdvanced('123', {fileUrl: 'fileUrl', tag: 'tag', etc: 'etc...'}, function(err,res){});
```

We're not a huge fan of overloading function parameters in javascript. It makes fairly messy functions at the top where we determine what's what.

**Comment by avbel** [comment]
I made `params` optional.
I.e. we can call `speakSentence` and `playAudio` as

``` js
// with callbacks
client.Bridge.playAudio(id, 'url', function(err){...});
client.Bridge.playAudio(id, 'url', {tag: 'tag'}, function(err){...});

// with promises
client.Bridge.playAudio(id, 'url').then(...);
client.Bridge.playAudio(id, 'url', {tag: 'tag'}).then(...);

```

Also I added function `playAudioAdavnced` which allows to pass all parameters as object.

``` js
client.Bridge.playAudioAdvanced(id, {fileUrl: 'url', tag: 'tag'}, function(err){...});
client.Bridge.playAudioAdvanced(id, {fileUrl: 'url', tag: 'tag'}).then(...);
```

**Comment by dtolb** [comment]
Oops sorry if I wasn't clear. With the addition of `playAudioAdvanced` and `speakSentenceAdvanced` we don't need to make the parameters optional on the `playAudio` & `speakSentence` methods. Since we have the `advanced` method to handle that. 

That way we simplify the logic on each of those functions and make it painfully clear to the end user what the intentions of each function are.

**Comment by avbel** [comment]
It is not clear for me now. Both `playAudio` and `speackSentence` make same request to Catapult API (POST /bridges/:id/audio). Difference in params only. What different will be in implementation of `playAudioAdvanced` and `speakSentenceAdvanced`? Current implementation `playAudioAdvanced`  put `params` as payload of Catapult Api request whitout any changes.

**Comment by dtolb** [comment]
@avbel sorry for the confusion. Because we have the `advanced` function now, we don't need to make optional parameters for `speakSentence` or `playAudioFile`.

I made some quick changes [here](https://github.com/bandwidthcom/node-bandwidth/pull/67) to demonstrate what I was talking about.

Take a look at the [examples](https://github.com/dtolb/node-bandwidth/blob/brige_update/docs/api.md#Bridge+speakSentence) section of the docs to get an idea of what I was going for.

**Comment by avbel** [comment]
Done. Removed optional params.

**Pull request merged**